src/client/auth/aws.rs

```diff
@@ -153,7 +174,41 @@ async fn authenticate_stream_inner(
     Ok(())
 }
 
+// Find credentials using MongoDB URI or AWS SDK
+pub async fn get_aws_credentials(credential: &Credential) -> Result<Credentials> {
```

Comment:
```suggestion
pub(crate) async fn get_aws_credentials(credential: &Credential) -> Result<Credentials> {
```

We generally only mark items `pub` when we intend for them to be part of the public API. `pub(crate)` will allow for internal usage in other parts of the driver.